### PR TITLE
sql/(parse,plan): implement describe queries

### LINKED ISSUE #216: Add support for basic query EXPLAIN
Explain query output has a really difficult to implement format: https://dev.mysql.com/doc/refman/8.0/en/explain-output.html

To be able to show some information to the user, we can do a small workaround and implement a specific, MySQL language compatible EXPLAIN statement format:

`[EXPLAIN|DESCRIBE|DESC] FORMAT=TREE [query]`

The output will be one column, one string row with the string tree representation of the generated logical plan:

```sql
mysql> EXPLAIN FORMAT=TREE SELECT * FROM repositories where repository_id != "blah";
+------------------------------------------------------------+
|                            plan                            |
+------------------------------------------------------------+
|Project(repositories.repository_id)                         |
| └─ PushdownProjectionAndFiltersTable                       |
|     ├─ Columns(repositories.repository_id)                 |
|     ├─ Filters(NOT(repositories.repository_id = "blah"))   |
|     └─ Table(repositories)                                 |
|         └─ Column(repository_id, TEXT, nullable=false)     |
+------------------------------------------------------------+
```

Maybe is worth it add more plans like `parsed`, `analyzed` and `optimized`.

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #216

### CODE REVIEW

File: engine_test.go [hunk 1]
```diff
@@ -491,6 +503,8 @@ func testQuery(t *testing.T, e *sqle.Engine, q string, r []sql.Row) {
 			rs = append(rs, row)
 		}
 
+		spew.Dump(rs)
```
Comment (ajnavarro) [inline engine_test.go]: leftover?

**Comment by erizocosmico** [comment]
Yep. Agh, every single time.
On Thu, 14 Jun 2018 at 17:43, Antonio Navarro Perez <notifications@github.com<mailto:notifications@github.com>> wrote:

@ajnavarro commented on this pull request.

________________________________

In engine_test.go<https://github.com/src-d/go-mysql-server/pull/222#discussion_r195472918>:

> @@ -491,6 +503,8 @@ func testQuery(t *testing.T, e *sqle.Engine, q string, r []sql.Row) {
                        rs = append(rs, row)
                }

+               spew.Dump(rs)


leftover?

**Comment by erizocosmico** [comment]
Fixed @ajnavarro

**Comment by ajnavarro** [comment]
@erizocosmico I think you need to rebase.

**Comment by erizocosmico** [comment]
Done

File: sql/parse/describe_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+func TestParseDescribeQuery(t *testing.T) {
+	testCases := []struct {
+		query  string
+		result sql.Node
+		err    *errors.Kind
+	}{
+		{
+			"DESCRIBE TABLE foo",
```
Comment (ajnavarro) [inline sql/parse/describe_test.go]: Describe table should work too. Is it still being parsed somewhere?

Comment (erizocosmico) [reply sql/parse/describe_test.go]: Yep, just not here

**Pull request merged**